src/game/AI/ScriptDevAI/scripts/northrend/naxxramas/boss_anubrekhan.cpp

```diff
@@ -21,127 +21,105 @@ SDComment: Intro text usage is not very clear. Requires additional research.
 
     SPELL_SELF_SPAWN_5          = 29105,                    // This spawns 5 corpse scarabs ontop of us (most likely the pPlayer casts this on death)
     SPELL_SELF_SPAWN_10         = 28864,                    // This is used by the crypt guards when they die
 
-    NPC_CRYPT_GUARD             = 16573
+    NPC_CRYPT_GUARD             = 16573,
+
+    SPELLSET_10N                = 1595601,
+    SPELLSET_25N                = 2924901,
 };
 
-static const DialogueEntry aIntroDialogue[] =
+enum AnubRekhanActions
 {
-    {SAY_GREET,   NPC_ANUB_REKHAN,  7000},
-    {SAY_TAUNT1,  NPC_ANUB_REKHAN,  13000},
-    {SAY_TAUNT2,  NPC_ANUB_REKHAN,  11000},
-    {SAY_TAUNT3,  NPC_ANUB_REKHAN,  10000},
-    {SAY_TAUNT4,  NPC_ANUB_REKHAN,  0},
-    {0, 0, 0}
+    ANUBREKHAN_ACTIONS_MAX,
+    ANUBREKHAN_SUMMON,
 };
 
-static const float aCryptGuardLoc[4] = {3333.5f, -3475.9f, 287.1f, 3.17f};
-
-struct boss_anubrekhanAI : public ScriptedAI
+struct boss_anubrekhanAI : public BossAI
 {
-    boss_anubrekhanAI(Creature* pCreature) : ScriptedAI(pCreature),
-        m_introDialogue(aIntroDialogue)
+    boss_anubrekhanAI(Creature* creature) : BossAI(creature, ANUBREKHAN_ACTIONS_MAX),
+        m_instance(static_cast<instance_naxxramas*>(creature->GetInstanceData())),
+        m_isRegularMode(creature->GetMap()->IsRegularDifficulty())
     {
-        m_pInstance = (instance_naxxramas*)pCreature->GetInstanceData();
-        m_bIsRegularMode = pCreature->GetMap()->IsRegularDifficulty();
-        m_introDialogue.InitializeDialogueHelper(m_pInstance);
-        m_bHasTaunted = false;
+        m_hasTaunted = false;
+        SetDataType(TYPE_ANUB_REKHAN);
+        AddOnAggroText(SAY_AGGRO1, SAY_AGGRO2, SAY_AGGRO3);
+        AddOnKillText(SAY_SLAY);
+        AddCustomAction(ANUBREKHAN_SUMMON, true, [&](){
+            DoCastSpellIfCan(nullptr, SPELL_SUMMON_GUARD);
+        });
         Reset();
-
-        DoCastSpellIfCan(m_creature, SPELL_DOUBLE_ATTACK, CAST_TRIGGERED | CAST_AURA_NOT_PRESENT);
     }
 
-    instance_naxxramas* m_pInstance;
-    DialogueHelper m_introDialogue;
-    bool m_bIsRegularMode;
-
-    uint32 m_uiImpaleTimer;
-    uint32 m_uiLocustSwarmTimer;
-    uint32 m_uiSummonTimer;
-    bool   m_bHasTaunted;
+    instance_naxxramas* m_instance;
+    bool m_isRegularMode;
+    bool m_hasTaunted;
 
     void Reset() override
     {
-        m_uiImpaleTimer      = 15000;
-        m_uiLocustSwarmTimer = 90000;
-        m_uiSummonTimer      = m_bIsRegularMode ? 20000 : 0;// spawn a guardian only after 20 seconds in normal mode; in heroic there are already 2 Guards spawned
+        m_creature->SetSpellList(m_isRegularMode ? SPELLSET_10N : SPELLSET_25N);
+        DoCastSpellIfCan(m_creature, SPELL_DOUBLE_ATTACK, CAST_TRIGGERED | CAST_AURA_NOT_PRESENT);
+    }
+
+    void Aggro(Unit *who) override
+    {
+        if (m_isRegularMode)
+            ResetTimer(ANUBREKHAN_SUMMON, 20s);
+        BossAI::Aggro(who);
     }
 
     void KilledUnit(Unit* pVictim) override
```

Comment:
If you can be bothered, get rid of the hungarian notation like pVictim in any script which you rewrite, so we have a clear progression of consistency. You are already nuking the whole file basically in terms of history, so this is a perfect opportunity to use normal camelCase notation instead. Same goes for instance script.